src/main/java/frc/robot/subsystems/DriveTrain.java

```diff
@@ -26,17 +33,23 @@ public class DriveTrain extends SubsystemBase {
   private ADIS16448_IMU m_imu;
 
   public DriveTrain() {
+    super(new PIDController(DriveTrainConstants.kP, DriveTrainConstants.kI, DriveTrainConstants.kD));
     m_leftFront = new WPI_TalonFX(DriveTrainConstants.kLeftFrontPort);
     m_leftRear = new WPI_TalonFX(DriveTrainConstants.kLeftRearPort);
     m_rightFront = new WPI_TalonFX(DriveTrainConstants.kRightFrontPort);
     m_rightRear = new WPI_TalonFX(DriveTrainConstants.kRightRearPort);
 
+    m_DriveTrainFeedforward = new SimpleMotorFeedforward(DriveTrainConstants.kS, DriveTrainConstants.kV);
+
     m_leftMotors = new MotorControllerGroup(m_leftRear, m_leftFront);
     m_rightMotors = new MotorControllerGroup(m_rightRear, m_rightFront);
 
     m_drive = new DifferentialDrive(m_leftMotors, m_rightMotors);
 
     m_imu = new ADIS16448_IMU();
+
+    getController().setTolerance(getSetpoint());
+    setSetpoint(DriveTrainConstants.kDistanceToTarget);
```

Comment:
Do not set a SetPoint inside the constructor, your SetPoint changes according to where you are on the field it's not constant.

For example, if I stand near the feeder, I will need to drive a whole lot more to the charge station than I would need if I were to stand near the grid.